src/squidpy/datasets/__init__.py

```diff
@@ -1,7 +1,55 @@
 from __future__ import annotations
 
-from squidpy.datasets._10x_datasets import visium, visium_hne_sdata
-from squidpy.datasets._dataset import *  # noqa: F403
-from squidpy.datasets._image import *  # noqa: F403
+from squidpy.datasets._datasets import (
+    # Type aliases for dataset names
+    AnnDataDatasets,
+    ImageDatasets,
+    SpatialDataDatasets,
+    VisiumDatasets,
+    # AnnData datasets
+    four_i,
+    imc,
+    merfish,
+    mibitof,
+    sc_mouse_cortex,
+    seqfish,
+    slideseqv2,
+    # 10x Genomics Visium
+    visium,
+    visium_fluo_adata,
+    visium_fluo_adata_crop,
+    # Image datasets
+    visium_fluo_image_crop,
+    visium_hne_adata,
+    visium_hne_adata_crop,
+    visium_hne_image,
+    visium_hne_image_crop,
+    visium_hne_sdata,
+)
 
-__all__ = ["visium", "visium_hne_sdata"]
+__all__ = [
+    # Type aliases
+    "VisiumDatasets",
+    "AnnDataDatasets",
+    "ImageDatasets",
+    "SpatialDataDatasets",
+    # 10x Genomics
+    "visium",
+    "visium_hne_sdata",
+    # AnnData datasets
+    "four_i",
+    "imc",
+    "seqfish",
+    "visium_hne_adata",
+    "visium_hne_adata_crop",
+    "visium_fluo_adata",
+    "visium_fluo_adata_crop",
+    "sc_mouse_cortex",
+    "mibitof",
+    "merfish",
+    "slideseqv2",
+    # Image datasets
+    "visium_fluo_image_crop",
+    "visium_hne_image_crop",
+    "visium_hne_image",
```

Comment:
nit, but don't fully understand the comments here? Everything visium_x is f.e. "10x Genomics". Either remove or more semantically useful split